operator/v1/types_network.go

```diff
@@ -123,6 +123,34 @@ type NetworkMigration struct {
 	// changing the MTU for the default network will be rejected.
 	// +optional
 	MTU *MTUMigration `json:"mtu,omitempty"`
+
+	// features contains the features migration configuration. Set this to migrate
+	// feature configuration when changing the cluster default network provider.
+	// if unset, the default operation is to migrate all the configuration of
+	// supported features.
+	// +optional
+	Features *FeaturesMigration `json:"features,omitempty"`
+}
+
+type FeaturesMigration struct {
+	// egressIP specifies whether or not the EgressIP configuration is migrated
+	// automatically when changing the cluster default network provider.
+	// If unset, this property defaults to 'true' and EgressIP configure is migrated.
+	// +optional
+	// +kubebuilder:default:=true
+	EgressIP bool `json:"egressIP,omitempty"`
+	// egressFirewall specifies whether or not the EgressFirewall configuration is migrated
+	// automatically when changing the cluster default network provider.
+	// If unset, this property defaults to 'true' and EgressFirewall configure is migrated.
+	// +optional
+	// +kubebuilder:default:=true
+	EgressFirewall bool `json:"egressFirewall,omitempty"`
+	// multicast specifies whether or not the multicast configuration is migrated
+	// automatically when changing the cluster default network provider.
```

Comment:
Does multicast work _during_ the migration? If not, it would be nice to mention that here.

Reply:
No, it doesn't. Neither does EgressIP or EgressFirewall since the current migration solution is a disruptive one. The target of feature migration in 4.12 is only to make sure the configuration of these features can be migrated automatically.